File: src/terceiro-bimestre/pfe/s18/a2/Livro.js
class Livro {
    constructor(id, titulo, autor, anoPublicacao, editora, genero) {
        this.__id = id; 
        this.__titulo = titulo;
        this.__autor = autor;
        this.__anoPublicacao = anoPublicacao;
        this.__editora = editora;
        this.__genero = genero;
    }

    getId() {
        return this.__id;
    }

    getTitulo() {
        return this.__titulo;
    }

    getAutor() {
        return this.__autor;
    }

    getAnoPublicacao() {
        return this.__anoPublicacao;
    }

    getEditora() {
        return this.__editora;
    }

    getGenero() {
        return this.__genero;
    }

    getInformacoes() {
        console.log(`Id: ${this.getId()}`);
        console.log(`Título: ${this.getTitulo()}`);
        console.log(`Autor: ${this.getAutor()}`);
        console.log(`Ano de Publicação: ${this.getAnoPublicacao()}`);
        console.log(`Editora: ${this.getEditora()}`);
        console.log(`Gênero: ${this.getGenero()}`);
    }
}
export default Livro;

File: src/quarto-bimestre/logica/biblioteca/Livro.py
class Livro:
    def __init__(self, titulo, autor, ano_publicacao, genero):
        self.titulo = titulo
        self.autor = autor
        self.ano_publicacao = ano_publicacao
        self.genero = genero
        self.disponivel = True

    def get_titulo(self):
        return self.titulo
    
    def get_autor(self):
        return self.autor
    
    def get_ano_publicacao(self):
        return self.ano_publicacao
    
    def get_genero(self):
        return self.genero
    
    def is_disponivel(self):
        return self.disponivel
    
    def mostrar_livro(self):
        print(f"Título: {self.get_titulo()}")
        print(f"Autor: {self.get_autor()}")
        print(f"Ano de Publicação: {self.get_ano_publicacao()}")
        print(f"Gênero: {self.get_genero()}")
        print(f"Disponível: {'Sim' if self.is_disponivel() else 'Não'}")
